src.common/Enums/Direction8.ts

```diff
@@ -1,14 +1,35 @@
-export interface Direction8 {
-	readonly id: number;
-	readonly name: string;
-	readonly x: number;
-	readonly y: number;
-	readonly nextCW: Direction8;
-	readonly nextCCW: Direction8;
-	readonly opposite: Direction8;
-}
+export class Direction8 implements Direction8 {
+	public static readonly UpLeft: Readonly<Direction8> = new Direction8(1, "upleft");
```

Comment:
For my benefit, what's the implications of using `static readonly` vs `const`?

Reply:
in TypeScript you can't have class-level constants. If you replaced `static readonly` with `const` it wouldn't compile. But it essentially does what `const` would do - create a staticly accessible variable that can't be changed.